### PR TITLE
NF: simplifying getCol

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Each time getCol is called, a new SystemTime is created and immediately deleted. That's not a real trouble, but still kind create useless object. 

This way, `getCol(Context, T)` will be called only when we actually need to add a Time object, when we don't expect the collection to exists.

This should also simplifies https://github.com/ankidroid/Anki-Android/pull/7005 .

### CODE REVIEW

**Comment by david-allison-1** [comment]
I was referring to `RobolectricTest.getCol()` when I mentioned it was hard to follow:

https://github.com/ankidroid/Anki-Android/blob/c2ee96822852dc89f297265ef62268e4cdb1dbbb/AnkiDroid/src/test/java/com/ichi2/anki/RobolectricTest.java#L176-L184

**Comment by Arthur-Milchior** [comment]
Oh okay.

I still believe this PR should be merged however, at least while Time is saved in Collection. Seems better to avoid creating objects and to be clear that the getCol with parameter is not expected to be used

**Comment by david-allison-1** [comment]
We really should be focusing on macro-optimisations unless we can prove that these changes make a substantial difference to execution time.

**Comment by Arthur-Milchior** [comment]
I'm not hoping to make any change in execution time. Hopefully, the time should be the same. My goal is to be able to do tests which does not have any time related problem. That is, which can simulate passing time to tests queues, and which can call background tasks without needing the extra work to need to wait for tasks

**Review by mikehardy — APPROVED** [review]
Yeah I think this does help testing because otherwise you get an old collection where you don't get the chance to alter SystemTime and lose our time-traveling-test-powers if I understand correctly
